### PR TITLE
Tool produces a 0-byte binary when compilation fails

### LINKED ISSUE #35: Tool produces a 0-byte binary when compilation fails
Steps:

1. Create a view with a compilation error (e.g. `int x = "Hello world";`)
2. Run `dotnet-razor-precompile` on the app with the right arguments.

Expected:
View compilation fails and no `.Precompiled.dll` binary is produced.

Actual:
View compilation fails and a 0-byte binary is produced.

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #35

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @Eilon 

Not sure if you'd prefer having a functional test to cover this.

File: src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Design/Internal/PrecompileRunCommand.cs [hunk 1]
```diff
@@ -151,15 +150,33 @@ private ResourceDescription[] GetResources(ViewCompilationInfo[] results)
             Directory.CreateDirectory(Path.GetDirectoryName(assemblyPath));
 
             EmitResult emitResult;
-            using (var assemblyStream = File.OpenWrite(assemblyPath))
+            using (var assemblyStream = new MemoryStream())
             {
-                using (var pdbStream = File.OpenWrite(Path.ChangeExtension(assemblyPath, ".pdb")))
+                using (var pdbStream = new MemoryStream())
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Design/Internal/PrecompileRunCommand.cs]: This is what we did in dnx as well 😄

**Comment by davidfowl** [comment]
Where's the test!

**Comment by Eilon** [comment]
Yeah, definitely need *some* kind of test.

**Comment by pranavkm** [comment]
🆙 📅

**Pull request closed without merging**